File: app/src/main/java/com/example/grocer21/MyFoodRecyclerViewAdapter.java
package com.example.grocer21;

import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.grocer21.Database.Food;
import com.example.grocer21.FoodFragment.OnListFragmentInteractionListener;

import java.util.List;
import java.util.Objects;

/**
 * {@link RecyclerView.Adapter} that can display a {@link Food} and makes a call to the
 * specified {@link OnListFragmentInteractionListener}.
 */
public class MyFoodRecyclerViewAdapter extends RecyclerView.Adapter<MyFoodRecyclerViewAdapter.ViewHolder> {

    private final LiveData<List<Food>> foodList;
    private final OnListFragmentInteractionListener mListener;

    public MyFoodRecyclerViewAdapter(LiveData<List<Food>> items, OnListFragmentInteractionListener listener) {
        foodList = items;
        mListener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.fragment_food, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder holder, int position) {
        holder.mItem = Objects.requireNonNull(foodList.getValue()).get(position);
        holder.mIdView.setText(foodList.getValue().get(position).getName());

        holder.mView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (null != mListener) {
                    mListener.onListFragmentInteraction(holder.mItem);
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return Objects.requireNonNull(foodList.getValue()).size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        final View mView;
        final TextView mIdView;
        Food mItem;

        ViewHolder(View view) {
            super(view);
            mView = view;
            mIdView = view.findViewById(R.id.item_number);
        }

        @NonNull
        @Override
        public String toString() {
            return super.toString() + " '"  + "'";
        }
    }
}

File: app/src/main/java/com/example/grocer21/MyFoodRecyclerViewAdapter.kt
package com.example.grocer21

import android.arch.lifecycle.LiveData
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.example.grocer21.FoodFragment.OnListFragmentInteractionListener
import com.example.grocer21.database.entities.Foods
import java.util.Objects

/**
 * [RecyclerView.Adapter] that can display a [Foods] and makes a call to the
 * specified [OnListFragmentInteractionListener].
 */
class MyFoodRecyclerViewAdapter(private val foodList: LiveData<List<Foods>>, private val mListener: OnListFragmentInteractionListener?) : RecyclerView.Adapter<MyFoodRecyclerViewAdapter.ViewHolder>() {

    private var foods = emptyList<Foods>()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.fragment_food, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.mItem = Objects.requireNonNull<List<Foods>>(foodList.value)[position]
        holder.mIdView.text = foodList.value!![position].getName()

        holder.mView.setOnClickListener {
            mListener?.onListFragmentInteraction(holder.mItem!!)
        }
    }

    internal fun setFoods(foods: List<Foods>) {
        this.foods = foods
        notifyDataSetChanged()
    }
    override fun getItemCount(): Int {
        return Objects.requireNonNull<List<Foods>>(foodList.value).size
    }

    inner class ViewHolder internal constructor(internal val mView: View) : RecyclerView.ViewHolder(mView) {
        internal val mIdView: TextView = mView.findViewById(R.id.item_number)
        internal var mItem: Foods? = null

        override fun toString(): String {
            return super.toString() + " '" + "'"
        }
    }
}
